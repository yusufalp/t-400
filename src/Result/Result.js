import React from 'react';
import { withRouter } from 'react-router-dom';

class Result extends React.Component {
  // user can start over with random question by clicking the start over button on the result page
  handleStartOver = () => {
    this.props.history.push('/')
  }
  render() {
    return (
      <div>
        <h2>Your results</h2>
        <p>You have {this.props.correct} correct questions and {this.props.incorrect} incorrect questions </p>
        <p>Your average score is {this.props.correct * 10}%</p>
        <h3>{`${this.props.correct === 10
          ? 'TERMINATOR'
          : 'GOOD JOB!'}`}</h3>
        <img className='result-img' src={`${this.props.correct === 10
          ? 'https://live.staticflickr.com/450/19869372178_6bf3dc0a10_b.jpg'
          : 'https://upload.wikimedia.org/wikipedia/commons/b/be/Arnold_Schwarzenegger_2%2C_2012.jpg'}`} alt='arnold' />
        <p>Will you be back?</p>
        <button type='submit' onClick={this.handleStartOver}>Do it again</button>
      </div>
    );
  }
}

export default withRouter(Result);