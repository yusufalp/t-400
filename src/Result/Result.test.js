import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Result from './Result';

describe('Result component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Result />
      </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Result />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})