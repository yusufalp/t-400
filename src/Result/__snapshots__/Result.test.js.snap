// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result component renders the UI as expected 1`] = `
<div>
  <h2>
    Your results
  </h2>
  <p>
    You have 
     correct questions and 
     incorrect questions 
  </p>
  <p>
    Your average score is 
    NaN
    %
  </p>
  <h3>
    GOOD JOB!
  </h3>
  <img
    alt="arnold"
    className="result-img"
    src="https://upload.wikimedia.org/wikipedia/commons/b/be/Arnold_Schwarzenegger_2%2C_2012.jpg"
  />
  <p>
    Will you be back?
  </p>
  <button
    onClick={[Function]}
    type="submit"
  >
    Do it again
  </button>
</div>
`;
